examples/graph/mpnn-molecular-graphs.py

```diff
@@ -0,0 +1,723 @@
+        Chem.SanitizeMol(molecule, sanitizeOps=Chem.SanitizeFlags.SANITIZE_ALL ^ flag)
+
+    Chem.AssignStereochemistry(molecule, cleanIt=True, force=True)
+    return molecule
+
+
+def graph_from_molecule(molecule):
+
+    # Initialize graph
+    atom_features = []
+    bond_features = []
+    pair_indices = []
+
+    for atom in molecule.GetAtoms():
+
+        atom_features.append(
+            atom_type(atom)
+            + num_hydrogens(atom)
+            + num_valence(atom)
+            + hybridization(atom)
+        )
+        # Add self-loop. Notice, this also helps against some edge cases where the
+        # last node has no edges. Alternatively, if no self-loops are used, for
+        # these edge cases, zero-padding on the output of the edge network is needed.
+        pair_indices.append([atom.GetIdx(), atom.GetIdx()])
+        bond_features.append([1.0] + [0.0, 0.0, 0.0, 0.0] + [0.0])
+
+        atom_neighbors = atom.GetNeighbors()
+
+        for neighbor in atom_neighbors:
+            bond = molecule.GetBondBetweenAtoms(atom.GetIdx(), neighbor.GetIdx())
+            pair_indices.append([atom.GetIdx(), neighbor.GetIdx()])
+            bond_features.append([0.0] + bond_type(bond) + is_conjugated(bond))
+
+    return atom_features, bond_features, pair_indices
+
+
+def graphs_from_smiles(smiles_list):
+
+    # Initialize graphs
+    atom_features_list = []
+    bond_features_list = []
+    pair_indices_list = []
+
+    for smiles in smiles_list:
+
+        molecule = molecule_from_smiles(smiles)
+
+        atom_features, bond_features, pair_indices = graph_from_molecule(molecule)
+
+        atom_features_list.append(atom_features)
+        bond_features_list.append(bond_features)
+        pair_indices_list.append(pair_indices)
+
+    # Convert lists to ragged tensors for tf.data.Dataset later on
+    return (
+        tf.ragged.constant(atom_features_list, dtype=tf.float32),
+        tf.ragged.constant(bond_features_list, dtype=tf.float32),
+        tf.ragged.constant(pair_indices_list, dtype=tf.int64),
+    )
+
+
+# Shuffle array of indices ranging from 0 to 2049
+permuted_indices = np.random.permutation(np.arange(df.shape[0]))
+
+# Train set: 80 % of data
+train_index = permuted_indices[: int(df.shape[0] * 0.8)]
+X_train = graphs_from_smiles(df.iloc[train_index].smiles)
+y_train = df.iloc[train_index].p_np
+
+# Valid set: 19 % of data
+valid_index = permuted_indices[int(df.shape[0] * 0.8) : int(df.shape[0] * 0.99)]
+X_valid = graphs_from_smiles(df.iloc[valid_index].smiles)
+y_valid = df.iloc[valid_index].p_np
+
+# Test set: 1 % of data
+test_index = permuted_indices[int(df.shape[0] * 0.99) :]
+X_test = graphs_from_smiles(df.iloc[test_index].smiles)
+y_test = df.iloc[test_index].p_np
+
+"""
+### Test out helper functions
+"""
+
+print(f"Name:\t{df.name[100]}\nSMILES:\t{df.smiles[100]}\nBBBP:\t{df.p_np[100]}")
+molecule = molecule_from_smiles(df.iloc[100].smiles)
+print("Molecule:")
+molecule
```

Comment:
Move this to the print statement

Reply:
The thing is, that the molecule will not be visualized if moved into the print statement (see Google Colab linked below)